lib/io/pwm.cpp

```diff
@@ -103,63 +138,24 @@ core::Result Pwm::setDutyCycleByTime(const std::uint32_t time_active)
   return core::Result::kSuccess;
 }
 
-core::Result Pwm::setPeriod(const std::uint32_t period)
+core::Result Pwm::setPeriod(const std::uint32_t period, const int period_file)
 {
-  // Avoid I/O if no change is required
-  if (current_period_ == period) {
-    logger_.log(core::LogLevel::kDebug, "PWM period is already set to %d, skipping I/O", period);
-    return core::Result::kSuccess;
-  }
   char write_buffer[10];
-  snprintf(write_buffer, sizeof(write_buffer), "%d", current_period_);
-  const auto num_bytes_written = write(period_file_, write_buffer, sizeof(write_buffer));
-  if (num_bytes_written != sizeof(write_buffer)) {
-    logger_.log(core::LogLevel::kFatal, "Failed to write to PWM period file");
-    return core::Result::kFailure;
-  }
-  logger_.log(core::LogLevel::kDebug, "Successfully set PWM period to %d", current_period_);
-  current_period_ = period;
+  snprintf(write_buffer, sizeof(write_buffer), "%d", period);
+  const auto num_bytes_written = write(period_file, write_buffer, sizeof(write_buffer));
```

Comment:
this is a non-optional return so should be explicitly typed, i believe it is an `int`